bin/compile-requirements.sh

```diff
@@ -12,8 +12,11 @@ export CUSTOM_COMPILE_COMMAND="$ make compile-requirements"
 # We need this installed, but we don't want it to live in the main requirements
 # We will need to periodically review this pinning
 
-pip install -U pip==23.1.2
-pip install pip-tools==6.13.0
+pip install -U pip==23.3.2
+pip install pip-tools==7.3.0
+
+# Drop the compiled reqs files, to help us pick up automatic subdep updates, too
+rm -f requirements/*.txt
```

Comment:
@robhudson Am I overlooking a benefit of keeping the previously compiled reqs files? With them present, pip-compile definitely picks up fewer changes, very commonly only the top-level changes and no natural/allowed subdep changes

Reply:
Not sure, but I like the idea of picking up more subdep changes so +1 if this helps make that happen.